tests/lifecycle/test_case_test.go

```diff
@@ -106,42 +89,12 @@ func (s serviceLifecycleTestCase) execute(resourceGroup string) error {
 		ResourceGroup:          resourceGroup,
 		Details:                serviceManager.GetEmptyInstanceDetails(),
 		ProvisioningParameters: s.provisioningParameters,
+		Parent:                 parent,
 	}
 
-	// Provision...
-	provisioner, err := serviceManager.GetProvisioner(plan)
-	if err != nil {
+	if _, err = s.provision(ctx, serviceManager, instance, plan); err != nil {
 		return err
 	}
-	stepName, ok := provisioner.GetFirstStepName()
-	// There MUST be a first step
-	if !ok {
-		return fmt.Errorf(
-			`Module "%s" provisioner has no steps`,
-			s.module.GetName(),
-		)
-	}
-	// Execute provisioning steps until there are none left
-	for {
-		var step service.ProvisioningStep
-		step, ok = provisioner.GetStep(stepName)
-		if !ok {
-			return fmt.Errorf(
-				`Module "%s" provisioning step "%s" not found`,
-				s.module.GetName(),
-				stepName,
-			)
-		}
-		instance.Details, err = step.Execute(ctx, instance, plan)
-		if err != nil {
-			return err
-		}
-		stepName, ok = provisioner.GetNextStepName(stepName)
-		// If there is no next step, we're done with provisioning
-		if !ok {
-			break
-		}
-	}
 
 	//Only test the binding operations if the service is bindable
 	if svc.GetBindable() {
```

Comment:
I debated refactoring the bind and deprovision functionality out, but right now it doesn't need to be reused so I left it in place. I think stylistically, execute() might read better if all of it was refactored so execute was really just executing subsequent methods that make up the tests and the blocks related to a given lifecycle are contained.